plugins/modules/ec2_instance.py

```diff
@@ -1967,6 +1972,10 @@ def main():
         ],
         supports_check_mode=True
     )
+
+    if not module.params.get('instance_type') and not module.params.get('launch_template'):
+        module.deprecate("Default value instance_type has been deprecated, in the future you must set an instance_type or a launch_template",
+                         date='2023-01-01', collection_name='amazon.aws')
```

Comment:
@jillr Just an info, do we use date or collection version for deprecations?

@mandar242 Other than that, it looks good to me. Thank you for taking time to work on this.

Reply:
@alinabuzachis date, we don't expect to always have the same cadence of major collection releases